core/src/test/scala/unit/kafka/coordinator/transaction/TransactionCoordinatorTest.scala

```diff
@@ -564,7 +564,7 @@ class TransactionCoordinatorTest {
       .anyTimes()
 
     val originalMetadata = new TransactionMetadata(transactionalId, producerId, producerId, (producerEpoch + 1).toShort,
-      producerEpoch, txnTimeoutMs, Ongoing, partitions, time.milliseconds(), time.milliseconds())
+      (producerEpoch - 1).toShort, txnTimeoutMs, Ongoing, partitions, time.milliseconds(), time.milliseconds())
```

Comment:
Hmm was this wrong? It seems weird to have last producer epoch set to a value which is 2 less than the producer epoch.

Reply:
Yeah, this was wrong. This and a couple other weird values were a consequence of that change above to how we set the last epoch in `endTransactions`. After fixing that and fixing some of the EasyMock returns, all the values make sense now (we don't have any last epoch's that are more than 1 less than the epoch).